src/hooks/usePyodide.tsx

```diff
@@ -400,10 +402,15 @@ let channelListenerRegistered = false;
 function ensureOpenChannelListener(pyodideProxy: PyodideProxy): void {
   if (channelListenerRegistered) return;
 
-  window.addEventListener("message", (event) => {
+  window.addEventListener("message", async (event) => {
```

Comment:
I'm fairly sure it's OK, though I can't point directly to anything in the spec that says so. As far as I can tell the spec just says it [needs to be a `Function` that takes an `Event` and doesn't return anything](https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-EventListener).

My understanding is that there can be an issue if you're using `Event` instance methods like `preventDefault()` or `stopPropagation()` (because they can come "too late" when after an `await`), but we're not using those here.

Since `async` and `await` are just sugar for `Promises`, we can fairly easily rewrite to avoid the `async` function, e.g.

```js
  window.addEventListener("message", (event) => {
    const msg = event.data;
    if (msg.type === "openChannel") {
      pyodideProxy.runPyAsync(`
        "${msg.appName}" in _shiny_app_registry
      `, { returnResult: "value" }).then((appExists) => {
        if (appExists) {
          pyodideProxy.openChannel(msg.path, msg.appName, event.ports[0]);
        }
      });
    }
  });
```

It's not as clean, but would you prefer I change it to that, so that we're absolutely sure?

Reply:
It looks like `pyodideProxy.openChannel()` is itself `async`, so even without this change I don’t think there’s any guarantee that the channel will be opened by the time the event handler returns.